src/coreclr/jit/indirectcalltransformer.cpp

```diff
@@ -990,15 +984,17 @@ class IndirectCallTransformer
             compiler->fgAddRefPred(remainderBlock, elseBlock);
             compiler->fgAddRefPred(elseBlock, checkBlock);
 
-            weight_t newWeight = checkBlock->bbWeight - thenBlock->bbWeight;
-            if (newWeight < 0)
+            // Calculate the likelihood of the else block as a remainder of the sum
+            // of all the other likelihoods.
+            unsigned elseLikelihood = 100;
+            for (uint8_t i = 0; i < origCall->GetInlineCandidatesCount(); i++)
             {
-                // There could be a small error leading to e.g. -0.00001 instead of 0.0 here
-                // Mostly because of inheritWeightPercentage for thenBlock;
-                newWeight = 0;
+                elseLikelihood -= origCall->GetGDVCandidateInfo(i)->likelihood;
             }
+            // Make sure it didn't overflow
+            assert(elseLikelihood <= 100);
 
-            elseBlock->setBBProfileWeight(newWeight);
+            elseBlock->inheritWeightPercentage(currBlock, elseLikelihood);
```

Comment:
For the multi-guess exact case this likelhood should end up at zero, right? We think we know all the possibilities.

Reply:
Yes, but the logic is shared with non-exact case where fallback may have non-zero likelihood (Dynamic PGO)

I plan to enable "no fallback needed" separately, I am thinking of doing something like "we're importing the last type-check and the call has GTF_M_EXACT_GDV flag --> convert the last pair of check-then to elseBock.